### DISCUSSION TITLE
Migrating from jenkins to github actions

### DISCUSSION THREAD

**Discussion by roshitho** [community/community#53668] [Actions] [answered]
### Select Topic Area

Question

### Body

I have a jenkins pipeline in groovy that I am migrating to github action.

The groovy script I am converting but at multiple places it is calling .sh files and .py files. Can I utilize these scripts directly in my github action run or is there any approach/best practice generally suggested?

Thanks for all the pointers (Note : I am aware of jenkins migration - https://github.com/orgs/community/discussions/categories/actions but this is more around best practices or guidance from those who have done it than a 1-1 mapping request.)

**Comment by airtower-luna** [comment]
You can call scripts in your repository from [`run` steps](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsrun), after checking out the repository. For Python you might want to use [`actions/setup-python`](https://github.com/actions/setup-python#setup-python) first, to make sure `python` is the interpreter version you want.

**Reply by roshitho** [reply]
Thank you. I can run it no issues, my concern was is it a best practice is there any context switch issue is there any better way etc.  Do people who migrates convert it to action or keep it as is in shell script (reusable script) ?

**Reply by airtower-luna** [reply, marked as answer]
I'd strongly recommend keeping the script files. *Technically* you could move the content of a script into a `run` string, set the [`shell` for the step](https://docs.github.com/en/actions/using-workflows/workflow-syntax-for-github-actions#jobsjob_idstepsshell) e.g. to Python, and run that way, but the only thing you'd "save" is a millisecond or two for the startup of the shell that runs your command otherwise. It'll be a lot easier to maintain your scripts if you keep them separate, you can easily run them locally if you need to, and if you ever want to switch CI again that'll be much easier, too. :slightly_smiling_face:
